app/lib/frontend/handlers/custom_api.dart

```diff
@@ -232,3 +239,36 @@ Future<shelf.Response> apiSearchHandler(shelf.Request request) async {
   }
   return jsonResponse(result, pretty: isPrettyJson(request));
 }
+
+/// Handles POST /api/packages/<package>/options
+Future<shelf.Response> packageOptionsHandler(
+    shelf.Request request, String package) async {
+  if (request.method.toUpperCase() == 'GET') {
+    final p = await backend.lookupPackage(package);
+    if (p == null) {
+      return notFoundHandler(request);
+    }
+    final options = PkgOptions(isDiscontinued: p.isDiscontinued);
+    return jsonResponse(options.toJson(), pretty: isPrettyJson(request));
+  } else if (request.method.toUpperCase() == 'PUT') {
+    try {
+      final body = await request.readAsString();
+      final options =
+          PkgOptions.fromJson(json.decode(body) as Map<String, dynamic>);
+      await backend.updateOptions(package, options);
+      return jsonResponse({'success': true}, pretty: false);
+    } on UnauthorizedAccessException catch (_) {
+      return jsonResponse({'error': 'Not Authorized.'},
+          status: 403, pretty: false);
+    } on ClientInputException catch (e) {
+      return jsonResponse({'error': e.message},
+          status: e.status, pretty: false);
+    } catch (e, st) {
+      _logger.warning('Error processing flag update.', e, st);
```

Comment:
We have a generic catch-all, but not something like this. It could become generic though (in a follow-up PR), e.g. if a request is going for `/api/` it should have a generic json response like the above (otherwise the generic html error page).